server/lib/processLogs.js

```diff
@@ -0,0 +1,97 @@
+const async = require('async');
+const request = require('request');
+const moment = require('moment');
+const loggingTools = require('auth0-log-extension-tools');
+
+const config = require('../lib/config');
+const logger = require('../lib/logger');
+
+module.exports = (storage) =>
+  (req, res, next) => {
+    const wtBody = (req.webtaskContext && req.webtaskContext.body) || req.body || {};
+    const wtHead = (req.webtaskContext && req.webtaskContext.headers) || {};
+    const isCron = (wtBody.schedule && wtBody.state === 'active') || (wtHead.referer === 'https://manage.auth0.com/' && wtHead['if-none-match']);
+
+    if (!isCron) {
+      return next();
+    }
+
+    const now = Date.now();
+
+    const sendLog = function (log, callback) {
+      if (!log) {
+        return callback();
+      }
+
+      const index = config('LOGSTASH_INDEX');
+      const data = {
+        post_date: now,
+        type_description: loggingTools.logTypes.get(log.type)
+      };
+
+      Object.keys(log).forEach((key) => {
+        data[key] = log[key];
+      });
+
+      data[index] = log[index] || 'auth0';
+      data.message = JSON.stringify(log);
+
+      const url = config('LOGSTASH_TOKEN') ? `${config('LOGSTASH_URL')}?token=${config('LOGSTASH_TOKEN')}` : config('LOGSTASH_URL');
+      const options = {
+        method: 'POST',
+        timeout: 20000,
+        url: url,
+        headers: { 'cache-control': 'no-cache', 'content-type': 'application/json' },
+        body: data,
+        json: true
+      };
+
+      if (config('LOGSTASH_USER') && config('LOGSTASH_PASSWORD')) {
+        options['auth'] = {
+          user: config('LOGSTASH_USER'),
+          pass: config('LOGSTASH_PASSWORD'),
+          sendImmediately: true
+        }
+      }
+
+      request(options, (err, resp, body) => {
+        const error = err || (body && body.error) || null;
+        callback(error);
+      });
+    };
+
+    const onLogsReceived = (logs, callback) => {
+      if (!logs || !logs.length) {
+        return callback();
+      }
+
+      logger.info(`Sending ${logs.length} logs to Logstash.`);
+
+      async.eachLimit(logs, 100, sendLog, callback);
+    };
+
+    const slack = new loggingTools.reporters.SlackReporter({ hook: config('SLACK_INCOMING_WEBHOOK_URL'), username: 'auth0-logs-to-logstash', title: 'Logs To Logstash' });
```

Comment:
Checking on line [96](https://github.com/auth0/auth0-logs-to-logstash/pull/4/files#diff-c7cebf846964f0fdc867268be3906c1cR96)